plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastoreClient.java

```diff
@@ -220,12 +222,23 @@ public Table getTableWithCapabilities(String databaseName, String tableName)
     }
 
     @Override
-    public void setTableColumnStatistics(String databaseName, String tableName, List<ColumnStatisticsObj> statistics)
+    public void setTableColumnStatistics(String databaseName, String tableName, List<ColumnStatisticsObj> statistics, AcidTransaction transaction)
             throws TException
     {
         ColumnStatisticsDesc statisticsDescription = new ColumnStatisticsDesc(true, databaseName, tableName);
-        ColumnStatistics request = new ColumnStatistics(statisticsDescription, statistics);
-        client.update_table_column_statistics(request);
+        ColumnStatistics columnStatistics = new ColumnStatistics(statisticsDescription, statistics);
+        if (transaction.isAnalyze()) {
```

Comment:
why only analyze? 
seems like correct behavior regardless why we have `transaction` here, so why not `if (!transaction.isTransactional())`?

Reply:
I was planning on enabling storing of stats for INSERT, DELETE and UPDATE in a separate PR, covered by issue #7190.  I've tested ANALYZE and I haven't tested INSERT and friends yet.  In any case, this can't now include INSERT, because that starts a transaction that doesn't get committed.  That's something to debug in the follow-on 🤔

Reply:
OTOH, you know the `else` path cannot work for transactional tables, so the condition here looks weird.

```suggestion
        if (transaction.isTransactional()) {
            verify(transaction.isAnalyze(), "Stats collection for transactional tables is currently supported with ANALYZE only");
```

Reply:
That's a good change.  Done.

> ```
> git rebase -i $(git merge-base master head)
> ```
> 
> 
> so that it's easier to follow up

I didn't know about `git merge-base`, but I think I always do the equivalent.

Reply:
please apply same change in `setPartitionColumnStatistics` below

Reply:
Done.